### PR TITLE
Migration trials: require credit card on all flows

### PR DESCRIPTION
**Pull request by p-jackson**
Related to https://github.com/Automattic/dotcom-forge/issues/6577

## Proposed Changes

* For the following migration trial flows, navigate to checkout to ensure credit card
   - `/setup/migration-signup`
   -  `/setup/site-migration`
   - `/setup/import-hosted-site`
   - `/setup/import-focused`

## Testing Instructions

1. Apply patch D146710-code
2. Enable store sandbox or use your own credit card
3. Test that you get a 7-day trial on all of the migration flows below:

 - **/setup/migration-signup**
      - Migrate a ninja site      
      - Choose trial
      - Verify zero amount checkout is shown and a card is required
  - **/setup/site-migration** and 
       - Create a free site then choose "Import exiting content or website" at `/setup/site-setup/goals`. 
       - Choose "Everything" which requires Creator
       - Choose trial
       - Verify zero amount checkout is shown and a card is required
  - **/setup/new-hosted-site**
       - Create a ninja website and migrate that
       - Choose the trial option
       - Verify zero amount checkout is shown and a card is required
   - **/setup/import-focused/sitePicker?from=[your ninja site slug]**
       - Choose the trial option
       - Verify zero amount checkout is shown and a card is required

  

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-upgrade-plan/index.tsx [hunk 1]
```diff
@@ -33,15 +38,19 @@ const SiteMigrationUpgradePlan: Step = function ( { navigation } ) {
...
 			onCtaClick={ () => {
 				navigation.submit?.( {
 					goToCheckout: true,
 					plan: plan.getPathSlug ? plan.getPathSlug() : '',
 				} );
 			} }
-			onFreeTrialSelectionSuccess={ () => {
-				navigation.submit?.( { freeTrialSelected: true } );
+			onFreeTrialClick={ () => {
+				// navigation.submit?.( { freeTrialSelected: true } );
```
Comment (mpkelly) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-upgrade-plan/index.tsx]: TODO remove this

**Comment by mpkelly** [comment]
Added @Automattic/serenity as a reviewer as I've modified some of the code they are working on. For context, our goal in this PR and wider project is to stop abuse of trial sites by requiring a credit card. As a result, we now go to the checkout when the user selects the trial. We are passing through `hosting_intent` as a product extra during checkout. I've rebased and updated the tests. Please double-check I didn't break something.

File: packages/shopping-cart/src/types.ts [hunk 2]
```diff
@@ -661,6 +661,7 @@ export interface RequestCartProductExtra extends ResponseCartProductExtra {
 	signup?: boolean;
 	headstart_theme?: string;
 	feature_slug?: string;
+	hosting_intent?: string;
```
Comment (sirbrillig) [inline packages/shopping-cart/src/types.ts:675]: I realize that none of these other properties have a description but I'd like to change that. Could you add a brief comment above this property to explain what it means?

Comment (mpkelly) [reply packages/shopping-cart/src/types.ts:675]: Good point! I have added a comment. I found the name is not very descriptive but it is only a short-term solution. This extra will be removed once we migrate all our flows to Migration Guru from Jetpack. 

https://github.com/Automattic/wp-calypso/pull/89839/commits/92f48a4c135baaae7de7a3e40eda3d8104a50dcd

**Comment by mpkelly** [comment]
Hi, @Automattic/caribou. Could we get some eyes on this as it modifies one of your flows. For context, we now require a credit card for trials. This PR and the linked diff do that for migration trials. The user now goes to the checkout page where we ensure a card is added.

**Comment by andres-blanco** [comment]
I've tested `/setup/site-migration` and it works as expected. 
My only concern is that choosing Free Trial and arriving to the checkout screen (even tough it's set to charge $0) is a little daunting and I think there's going to be drop-off.

Maybe, it would be less unexpected if we add a message of some kind to indicate that we will be requiring credit card verification in this screen
image

**Comment by markbiek** [comment]
We also have `/setup/migration-signup` which is a streamlined version of `/setup/site-migration`.

Should this change also apply to `/setup/migration-signup`?

**Comment by mpkelly** [comment]
>My only concern is that choosing Free Trial and arriving to the checkout screen (even tough it's set to charge $0) is a little daunting and I think there's going to be drop-off.

Thanks, @andres-blanco. I had the same concerns, and we created ticket https://github.com/Automattic/dotcom-forge/issues/6637 for this. I think we should make it clear that you will never be billed somewhere. 

>We also have /setup/migration-signup
TIL. We have a lot of migration flows! I will need to look at this today @markbiek. 

Thanks for the reviews, Serenity!

**Review by ouikhuan — COMMENTED** [review]
I've tested on both flows:

- /setup/import-hosted-site
- /setup/import-focused

And the flow works as expected, nice work! However, when looking into the logs by apply the diff ( D146710-code ), I noticed the trial site is not created with utf-8 charset when starting from the `http://calypso.localhost:3000/setup/import-focused/import?siteSlug=<site-slug>`.

**Comment by mpkelly** [comment]
> I noticed the trial site is not created with utf-8 charset when starting from the http://calypso.localhost:3000/setup/import-focused/import?siteSlug=<site-slug>

Thanks, @ouikhuan. This is expected for now as UTF-8 is only used on flows using Migration Guru.

**Comment by mpkelly** [comment]
I made another commit so that `hosting_intent=migrate` is sent on flow `/setup/migration-signup`. If the build passes, I will merge this. 

Screenshot 2567-04-30 at 09 01 05

**Comment by mpkelly** [comment]
Thanks for the reviews. This has been merged.

**Pull request merged**